implementations/rust/ockam/ockam_vault_aws/src/vault/aws_kms_client.rs

```diff
@@ -143,22 +150,24 @@ impl AwsKmsClient {
         Err(Error::UnsupportedKeyType.into())
     }
 
-    pub(crate) async fn compute_key_id(&self, public_key: &PublicKey) -> Result<KeyId> {
-        log::trace!(%public_key, "compute the key id for a public key");
-        let output = self
-            .client
-            .describe_key()
-            .key_id(public_key.to_string())
-            .send()
-            .await
-            .map_err(|err| {
-                log::error!(%public_key, %err, "failed to get the key description");
-                Error::MissingKeyId
-            })?;
-        if let Some(metadata) = output.key_metadata() {
-            log::debug!(%public_key, "received key id");
-            if let Some(key_id) = metadata.key_id() {
-                return Ok(key_id.to_string());
+    /// Return the key id corresponding to a public key from the KMS
+    /// This function is particularly inefficient since it lists all the keys
+    /// This is why there is a cache in the AwsKms module to avoid this call
+    pub(crate) async fn get_key_id(&self, public_key: &PublicKey) -> Result<KeyId> {
+        let output = self.client.list_keys().send().await.map_err(|err| {
+            log::error!(%public_key, %err, "failed to list all keys");
+            Error::MissingKeys
+        })?;
+
+        if let Some(keys) = output.keys() {
+            log::debug!(%public_key, "received keys");
+            for key in keys {
+                if let Some(key_id) = key.key_id() {
+                    let one_public_key = self.public_key(&key_id.to_string()).await?;
+                    if one_public_key == public_key.clone() {
```

Comment:
Is Clone really necessary here?

Reply:
Removed

Reply:
Could we do? :D
```
if &one_public_key == public_key {
```

Reply:
Yeah, but isn't that causing an excess data copy? Which isn't much, but why don't we just use references/

Reply:
`one_public_key == *public_key` is causing data copy?

Reply:
I had an impression that `*public_key` creates a copy

Reply:
It seems that both are the same but (according to at least one person on StackOverflow) comparing references is more frequent: https://stackoverflow.com/questions/74070545/comparing-a-value-and-a-reference. So I changed to that version.